### PR TITLE
Optimize the controllers flag

### PR DESCRIPTION
**Pull request by iawia002**
Signed-off-by: Xinzhao Xu <z2d@jifangcheng.com>

**What type of PR is this?**

/kind cleanup

This PR uses the `StringSliceVar` method to optimize the `--controllers` flag, this will make the original code easier to use and read.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref #1083

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @RainbowMango @snowplayfire

### CODE REVIEW

File: cmd/controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -164,7 +161,7 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 		}
 	}
 
-	if IsControllerEnabled("clusterStatus", controllers) {
+	if opts.IsControllerEnabled("clusterStatus") {
```
Comment (iawia002) [inline cmd/controller-manager/app/controllermanager.go:164]: I have one more question, should we use the name format used by Kubernetes:

```
All controllers: attachdetach, bootstrapsigner, cloud-node-lifecycle
```

ref: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/

Comment (RainbowMango) [reply cmd/controller-manager/app/controllermanager.go:164]: @snowplayfire How do you think?

Comment (RainbowMango) [reply cmd/controller-manager/app/controllermanager.go:164]: I'm ok with both formats (Linux or camel).

**Comment by RainbowMango** [comment]
No behavior changes.
/lgtm
/approve

**Pull request merged**